extract.py

```diff
@@ -120,6 +143,87 @@ def extract_database(args):
     return output_rows
 
 
+def extract_csv(args, mapping=None, init_id=0, date_format="%y%m%d", reg_rules=None):
+    if mapping is None:
+        mapping = dict()
+    if reg_rules is None:
+        reg_rules = dict()
+    translation = {"meta": {"reg_rules": reg_rules}, "mapping": mapping}
+    translation["meta"]["date_format"] = date_format
+    output_rows = []
+    report = get_report()
+    person_id = init_id
+    with open(args.source, "r") as datasource:
+        rows = csv.DictReader(datasource)
+        for row in rows:
+            handled_row = translate_row(row, report, translation)
+            if not handled_row[0]:
+                handled_row[0] = person_id
+                person_id += 1
+            output_rows.append(handled_row)
+
+    shuffle(output_rows)
```

Comment:
What's the reason for the shuffle?

Reply:
That's part of the original code that was cloned into this CSV handling function. It's protection against potentially leaking any info about individuals based on the sequence of hashes sent to the linkage agent